src/containers/habit-tracker/habits/Habits.tsx

```diff
@@ -194,11 +194,38 @@ export function Habits() {
   // Handler for deleting a habit
   const handleDeleteHabit = (index: number) => {
     if (window.confirm("Are you sure you want to delete this habit?")) {
+      const habitToDelete = habits[index].name;
       const updatedHabits = habits.filter((_, idx) => idx !== index);
+
+      // Check if the deleted habit is the selected habit
+      const selectedHabit = localStorage.getItem(LocalStorage.SELECTED_HABIT);
+
+      if (selectedHabit && selectedHabit === habitToDelete) {
+        localStorage.removeItem(LocalStorage.SELECTED_HABIT);
+      }
+
       updateHabits(updatedHabits, setHabits);
     }
   };
 
+  // Handler for deleting all habits and logs
+  const handleDeleteAllHabits = () => {
+    if (
+      window.confirm("Are you sure you want to delete all habits and logs?")
+    ) {
+      updateHabits([], setHabits);
+      setHabitsLog({});
+      localStorage.removeItem(LocalStorage.HABITS);
+      localStorage.removeItem(LocalStorage.HABITS_LOG);
+
+      const selectedHabit = localStorage.getItem(LocalStorage.SELECTED_HABIT);
+
+      if (selectedHabit) {
+        localStorage.removeItem(LocalStorage.SELECTED_HABIT);
```

Comment:
not sure if this check is required. 
If you are deleting all the habits, simply just remove the `LocalStorage.SELECTED_HABIT` as well. Hopefully, it won't throw an error.